libs/blocks/icon-block/icon-block.js

```diff
@@ -54,14 +54,40 @@ function decorateContent(el) {
   if (text) {
     text.classList.add('text-content');
     const image = block.querySelector(':scope img');
-    if (image) image.closest('p').classList.add('icon-area');
-    // place standalone links inside an action-area
     const lastElem = text.lastElementChild;
     if (lastElem.children.length === 1
       && lastElem.lastElementChild.nodeName === 'A'
       && lastElem.lastElementChild.innerText === lastElem.innerText) {
       text.lastElementChild.classList.add('action-area');
     }
+
+    if (image) {
+      const iconP = image.closest('p');
+      iconP.classList.add('icon-area');
+      const iconLink = iconP.querySelector('a');
+      const actionLink = lastElem?.children.length === 1 && lastElem.lastElementChild.nodeName === 'A'
+        ? lastElem.lastElementChild
+        : null;
+
+      if (iconLink && actionLink) {
+        const wrapper = createTag('a', {
+          href: actionLink.href,
+          class: 'wrapper-anchor',
+        });
+
+        iconLink.replaceWith(...iconLink.childNodes);
+
+        const actionContent = actionLink.innerHTML;
+        lastElem.innerHTML = actionContent;
```

Comment:
Maybe there is no need for variable 
```suggestion
        lastElem.innerHTML = actionLink.innerHTML;
```